client/src/App.tsx

```diff
@@ -23,14 +21,14 @@ function App({ socket }: Props) {
       const { data } = await serverAxios.get("/watch-app/user");
       setAuth(data);
     };
-    getUserId();
-  }, [setAuth]);
 
-  useEffect(() => {
-    socket.on(RoomEvent.SERVER_ROOMS, ({ rooms }) => {
-      getRooms && getRooms(rooms);
-    });
-  }, [getRooms, socket]);
+    const fetchRooms = async () => {
```

Comment:
Implementation wise: 

You should make `fetchRooms` and `getUserId` properties of the App component and pass it into the component. If you define the methods like this, your implementation will bind to an exact server endpoint. This does not great for testing as you cannot "mock" the side effect results.

Reply:
a bit confused. Do you mean we should put these methods in a new file and then pass to the App component?